Fix Extension Docs (#2377)
kontena/lens: docs/extensions/testing-and-publishing/testing.md

## Renderer Process Unit Testing

UI components in the extension's renderer process are based on React/ReactDOM.
These components can be tested by popular React testing tools like [React Testing Library](https://github.com/testing-library/react-testing-library).

> If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold extension project then the testing environment for render process is already setup for you.

Reviewer: ```suggestion
If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold extension project then the testing environment for render process is already set up for you.
```

## Renderer Process Unit Testing

> UI components in extension renderer process are based on React/ReactDOM.

Reviewer: ```suggestion
UI components in the extension's renderer process are based on React/ReactDOM.
```

These components can be tested by popular React testing tools like [React Testing Library](https://github.com/testing-library/react-testing-library).

> If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold extension project.
> The testing environment for render process are already setup for you.

Reviewer: ```suggestion
If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold your extension project then the testing environment for the renderer process is already set up for you.
```

In the example we used [React Testing Library](https://github.com/testing-library/react-testing-library) but any React testing framework can be used to test renderer process UI components.

There are more example tests in the generator's [template](https://github.com/lensapp/generator-lens-ext/tree/main/generators/app/templates/ext-ts/components).
Extend your tests based on the examples.

## Main Process Unit Testing

> Code in the extension main process are just normal JavaScript files that has access to extension api, you can write unit tests using any testing framework.

Reviewer: ```suggestion
Code in the extension's main process consists of normal JavaScript files that have access to extension api, you can write unit tests using any testing framework.
```

If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold your extension project.

> The testing environment [Jest](https://jestjs.io/) are setup for you.

Reviewer: ```suggestion
If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold your extension project then the [Jest](https://jestjs.io/) testing environment is set up for you.
```

## Main Process Unit Testing

Code in the extension's main process consists of normal JavaScript files that have access to extension api, you can write unit tests using any testing framework.

> If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold your extension project then the [Jest](https://jestjs.io/) is setup for you.

Reviewer: ```suggestion
If you are using the [Yeoman Lens Extension Generator](https://github.com/lensapp/generator-lens-ext) to scaffold your extension project then the [Jest](https://jestjs.io/) testing environment is set up for you.
```
